Nasty little hack to preserve NotImplementedError raised in interpreter

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #53973 Restore storage on meta tensors; increase meta coverage
* #54711 Compute Tensor::toString() without reference to backend
* #54710 Make memory overlap a little less precise so it works with null data ptr
* #54709 Make the fuser raise NotImplementedError when unknown device is hit
* #54708 Avoid testing device in cdist when called in a "Math" context
* #54707 Make convolution_overrideable default implementation raise NotImplementedError
* #54706 Disambiguate where 'Doesn't run' error message comes from
* #54705 Fix missing format string qualifier
* #54704 Rename linalg.det test so that it generates a valid method name
* #54703 Add empty_generic, which allocates an empty tensor in a device-generic way
* #54597 Fix some more native_functions.yaml mistakes
* **#54627 Nasty little hack to preserve NotImplementedError raised in interpreter**

This is the simplest little fix to get interpreter to preserve
NotImplementedError, so that the test suite doesn't start choking
on meta tensors not working in interpreter.  It is sound and correct
but doesn't work for other c10::Error subclasses with special handling.
A more proper fix is requested at
https://github.com/pytorch/pytorch/issues/54612

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27328666](https://our.internmc.facebook.com/intern/diff/D27328666)

ezyang committed:
Nasty little hack to preserve NotImplementedError raised in interpreter

This is the simplest little fix to get interpreter to preserve
NotImplementedError, so that the test suite doesn't start choking
on meta tensors not working in interpreter.  It is sound and correct
but doesn't work for other c10::Error subclasses with special handling.
A more proper fix is requested at
https://github.com/pytorch/pytorch/issues/54612

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

mruberry reviewed (approved):
Sometimes exceptions must be made for exceptions.

The clang-tidy failure looks real, however.

ezyang committed:
Update on "Nasty little hack to preserve NotImplementedError raised in interpreter"


This is the simplest little fix to get interpreter to preserve
NotImplementedError, so that the test suite doesn't start choking
on meta tensors not working in interpreter.  It is sound and correct
but doesn't work for other c10::Error subclasses with special handling.
A more proper fix is requested at
https://github.com/pytorch/pytorch/issues/54612

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D27328666](https://our.internmc.facebook.com/intern/diff/D27328666)

[ghstack-poisoned]
